observer.go

```diff
@@ -39,50 +43,27 @@ type observer struct {
 	observers []Observer
 }
```

Comment:
The `observer` is probably needed since, it implements the `Observer` interface. If we remove this struct `observer`, the function `OnStartSpan(operationName string, options opentracing.StartSpanOptions)` i.e., without the span argument, can't be defined which in turn will affect the existing external observers, right? Is there any other way we can replace this?